HomeWork2/Maksim_Panibrat_HW2_optional.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+
+void Christmas_Tree() {
+    std::cout << "This program prints christmas tree from stars\n"
+        "Enter tree size (max number of stars):  ";
+    size_t max_stars_number;
+    std::cin >> max_stars_number;
+    if (max_stars_number % 2 == 0) {
+        max_stars_number += 1;
+    }
+    char star = '*', null = ' ';
+    const size_t half_max_stars = max_stars_number / 2;
+    for (size_t string_number = 0; string_number < half_max_stars + 1;
+        ++string_number) {
+        for (size_t i = 0; i < max_stars_number; i++) {
+            if (i < (half_max_stars - string_number) ||
+                i >(half_max_stars + string_number)) {
+                std::cout << null;
+            }
+            else {
+                std::cout << star;
+            }
+        }
+        std::cout << std::endl;
+    }
+}
+
+void Bit_Counter() {
+    size_t check = 131072;
+    std::cout << "This programm counts set bits in range [0..."<<check<<"].\n";
+    size_t input;
+    short set_bits{};
+    std::cout << "Enter number for analize: ";
+    for (std::cin >> input; check != 0; check = check >> 1) {
+        if ((input & check) != 0) {
+            set_bits++;
+        }
+    }
```

Comment:
```suggestion
   size_t size_bit{sizeof(input) * 8};
   size_t current_index_bit{};
    for (std::cin >> input; current_index_bit != size_bit; ++current_index_bit) {
        if ((input & 1 << current_index_bit) != 0) {
            set_bits++;
        }
    }
```
наверное как то так, не проверял код. Но точно не то что вы написали 131072